tfhe/src/core_crypto/algorithms/lwe_linear_algebra.rs

```diff
@@ -311,20 +386,51 @@ pub fn lwe_ciphertext_plaintext_sub_assign<Scalar, InCont>(
 ) where
     Scalar: UnsignedInteger,
     InCont: ContainerMut<Element = Scalar>,
+{
+    let ciphertext_modulus = lhs.ciphertext_modulus();
+    if ciphertext_modulus.is_compatible_with_native_modulus() {
+        lwe_ciphertext_plaintext_sub_assign_native_mod_compatible(lhs, rhs);
+    } else {
+        lwe_ciphertext_plaintext_sub_assign_other_mod(lhs, rhs);
+    }
+}
+
+pub fn lwe_ciphertext_plaintext_sub_assign_native_mod_compatible<Scalar, InCont>(
+    lhs: &mut LweCiphertext<InCont>,
+    rhs: Plaintext<Scalar>,
+) where
+    Scalar: UnsignedInteger,
+    InCont: ContainerMut<Element = Scalar>,
 {
     let body = lhs.get_mut_body();
     let ciphertext_modulus = body.ciphertext_modulus();
     assert!(ciphertext_modulus.is_compatible_with_native_modulus());
-    if ciphertext_modulus.is_native_modulus() {
-        *body.data = (*body.data).wrapping_sub(rhs.0);
-    } else {
-        *body.data = (*body.data).wrapping_sub(
-            rhs.0
-                .wrapping_mul(ciphertext_modulus.get_power_of_two_scaling_to_native_torus()),
-        );
+    match ciphertext_modulus.kind() {
+        CiphertextModulusKind::Native => *body.data = (*body.data).wrapping_sub(rhs.0),
+        CiphertextModulusKind::NonNativePowerOfTwo => {
```

Comment:
Could factor `*body.data = (*body.data).wrapping_sub(` out of the `match`

Reply:
same as above